src/main/java/redis/clients/jedis/CacheJedis.java

```diff
@@ -0,0 +1,265 @@
+package redis.clients.jedis;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import redis.clients.jedis.util.SafeEncoder;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLParameters;
+import javax.net.ssl.SSLSocketFactory;
+import java.net.URI;
+import java.util.List;
+
+import static redis.clients.jedis.Protocol.Command.CLIENT;
+
+public class CacheJedis extends Jedis {
+    final Jedis jedisSub = new Jedis(); // Jedis instance for Client tracking
+    final String clientId= String.valueOf(jedisSub.clientId());
+    final LoadingCache<String, Object> cache = CacheBuilder.newBuilder()
+            .build(new CacheLoader<String, Object>() {
+                @Override public Object load(String key) throws Exception {
+                    return new Object();
+                }
+            });
+    final JedisPubSub jedisPubSub = new JedisPubSub() {
+        //Overriding different methods of pub sub to take appropriate actions
+        @Override
+        public void onMessage(String channel, List<Object> message) {
+            System.out.println("Channel " + channel + " has sent a message : " + message);
```

Comment:
1. remove these in main code 
2. add logger support - check slf4j and add loggers here